### PR TITLE
Added support for dashes in setting names

### LINKED ISSUE #60: Support dash in setting names
Since the flags augmented by the transitions are converted to attributes, and dash characters (`-`) aren't supported in attributes, it is impossible to use with_cfg on settings which are located in a path that contains a dash.

This should probably be supported somehow.

### PR DESCRIPTION
**Pull request by Swey9**
- Added tests to ensure the validity conversion of setting.bzl

Fixes #60

### CODE REVIEW

File: with_cfg/private/setting.bzl [hunk 1]
```diff
@@ -9,10 +9,11 @@ def validate_and_get_attr_name(setting):
...
 
-        # Ensure that the hash, which is a (signed) 32-bit integer, is non-negative, so that it does
-        # not contain a dash, which is not allowed in attribute names. Also ensure that the
-        # attribute name starts with a letter as it needs to be a valid identifier.
-        return "s_{}_{}".format(hash(str(setting)) + 2147483648, setting.name)
+        # Ensure the attribute name starts with a letter and doesn't contain a dash as it needs
+        # to be a valid identifier. Achieves that by ensuring that the hash, which is a (signed)
+        # 32-bit integer, is non-negative, so that it doesn't contain a dash, and in addition,
+        # replacing any potential dashes in the setting name.
+        return "s_{}_{}".format(hash(str(setting)) + 2147483648, setting.name.replace("-", "_"))
```
Comment (fmeum) [inline with_cfg/private/setting.bzl:16]: There could be other characters in target names that aren't allowed in attributes. 

How about using the function you wrote for the test to loop over the characters and replace everything that isn't alphanumeric with an underscore? That should deal with any issues of this kind.

File: with_cfg/private/setting.bzl [hunk 2]
```diff
@@ -3,16 +3,31 @@ load(":select.bzl", "map_attr")
...
     if is_label(setting):
         # Trigger an early error if the label refers to an invalid repo name.
         # buildifier: disable=no-effect
         setting.workspace_name
 
-        # Ensure that the hash, which is a (signed) 32-bit integer, is non-negative, so that it does
-        # not contain a dash, which is not allowed in attribute names. Also ensure that the
-        # attribute name starts with a letter as it needs to be a valid identifier.
-        return "s_{}_{}".format(hash(str(setting)) + 2147483648, setting.name)
+        return make_valid_identifier("s_{}_{}".format(hash(str(setting)), setting.name))
```
Comment (fmeum) [inline with_cfg/private/setting.bzl:30]: Could you move the `s_` prefix into `make_valid_identifier` and bring back the last sentence in the comment for it?

**Review by fmeum — COMMENTED** [review]
Sorry for the delay, I didn't see your update. 

Please just ping me or rerequest a review after pushing a change!

**Comment by Swey9** [comment]
> Sorry for the delay, I didn't see your update. 
> 
> Please just ping me or rerequest a review after pushing a change!

Yeah sure, will do that tomorrow :) thank you

**Review by fmeum — APPROVED** [review]

**Pull request merged**